src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -317,6 +317,22 @@ public bool IsBaseConstructorInitializer(SyntaxToken token)
         public bool IsQueryExpression(SyntaxNode node)
             => node.Kind() == SyntaxKind.QueryExpression;
 
+        public bool IsQueryKeyword(SyntaxToken token)
+            => token.IsKind(
+                SyntaxKind.FromKeyword,
+                SyntaxKind.GroupKeyword,
+                SyntaxKind.JoinKeyword,
+                SyntaxKind.IntoKeyword,
+                SyntaxKind.LetKeyword,
+                SyntaxKind.ByKeyword,
+                SyntaxKind.SelectKeyword,
+                SyntaxKind.OrderByKeyword,
+                SyntaxKind.OnKeyword,
+                SyntaxKind.EqualsKeyword,
+                SyntaxKind.AscendingKeyword,
+                SyntaxKind.DescendingKeyword) ||
+            (token.IsKind(SyntaxKind.InKeyword) && (token.Parent?.IsKind(SyntaxKind.FromClause, SyntaxKind.JoinClause, SyntaxKind.JoinIntoClause) == true));
```

Comment:
:bulb: Use `token.IsParentKind` instead of `token.Parent?.IsKind`.

Reply:
> Use token.IsParentKind

Couldn't find such a method. There seems to be only a `IsParentKind(this SyntaxNode node` extension method.